src/generic_factory.h

```diff
@@ -857,9 +857,17 @@ inline bool one_char_symbol_reader( const JsonObject &jo, const std::string &mem
         return false;
     }
     if( sym_as_string.size() != 1 ) {
-        jo.throw_error( member_name + " must be exactly one ASCII character", member_name );
+        jo.throw_error(
+            member_name + " must be exactly one ASCII character but was %zu characters",
```

Comment:
`JsonObject::throw_error` does not do string formatting. You have to build the string yourself.
```suggestion
            member_name, sym_as_string.size() );
```
```suggestion
            member_name + " must be exactly one ASCII character but was " + std::to_string( sym_as_string.size() ) + " characters",
```

Same below.

Reply:
Thanks.  It's not the first time I've made that mistake...